extension/src-language-server/fta/diagram/fta-layout-config.ts

```diff
@@ -17,19 +17,79 @@
-    protected graphOptions(_sgraph: SGraph, _index: SModelIndex): LayoutOptions {
-        return {
+    protected graphOptions(sgraph: FTAGraph, _index: SModelIndex): LayoutOptions {
+        const options: LayoutOptions = {
             "org.eclipse.elk.direction": "DOWN",
-            "org.eclipse.elk.layered.nodePlacement.strategy": "NETWORK_SIMPLEX",
+            "org.eclipse.elk.layered.nodePlacement.strategy": "BRANDES_KOEPF",
+            "org.eclipse.elk.portConstraints": "FIXED_SIDE",
+            "org.eclipse.elk.hierarchyHandling": "INCLUDE_CHILDREN",
+            "org.eclipse.elk.spacing.portPort": "0.0",
+            "org.eclipse.elk.layered.nodePlacement.bk.fixedAlignment": "BALANCED"
         };
+
+        if (sgraph.modelOrder) {
+            options["org.eclipse.elk.layered.considerModelOrder.strategy"] = "NODES_AND_EDGES";
+            options["org.eclipse.elk.layered.crossingMinimization.forceNodeModelOrder"] = "true";
+            options["org.eclipse.elk.separateConnectedComponents"] = "false";
+        }
+        return options;
     }
 
-    protected nodeOptions(_snode: SNode, _index: SModelIndex): LayoutOptions | undefined {
-        return {
+    protected nodeOptions(snode: SNode, _index: SModelIndex): LayoutOptions | undefined {
+        const options: LayoutOptions = {
+            "org.eclipse.elk.portConstraints": "FIXED_SIDE",
+            "org.eclipse.elk.spacing.portPort": "0.0",
             "org.eclipse.elk.nodeLabels.placement": "INSIDE V_CENTER H_CENTER",
         };
+        switch (snode.type) {
+            case FTA_NODE_TYPE:
+                switch ((snode as FTANode).nodeType) {
+                    case FTNodeType.PARENT:
+                        options["org.eclipse.elk.direction"] = "DOWN";
+                        options["org.eclipse.elk.padding"] = "[top=0.0,left=0.0,bottom=10.0,right=0.0]";
+                        options["org.eclipse.elk.layered.spacing.nodeNodeBetweenLayers"] = "2";
+                        options["org.eclipse.elk.hierarchyHandling"] = "INCLUDE_CHILDREN";
+                        break;
+                    case FTNodeType.COMPONENT:
+                    case FTNodeType.CONDITION:
+                        options["org.eclipse.elk.nodeSize.constraints"] = "MINIMUM_SIZE, NODE_LABELS";
+                        options["org.eclipse.elk.nodeSize.minimum"] = "(30, 30)";
+                        options["org.eclipse.elk.spacing.labelNode"] = "20.0";
+                        break;
+                }
+                break;
+            case FTA_DESCRIPTION_NODE_TYPE:
+                options["org.eclipse.elk.nodeSize.constraints"] = "NODE_LABELS";
+                break;
+        }
+        return options;
+    }
+
+    protected portOptions(sport: FTAPort, index: SModelIndex): LayoutOptions | undefined {
+        if (sport.type === FTA_PORT_TYPE) {
+            let side = "";
+            switch ((sport as FTAPort).side) {
```

Comment:
I didn't test this but I think you can replace the whole switch case with this:
`let side= PortSide[(sport as FTAPort).side];`